### ISSUE TITLE
[Bug] overlap causes performance loss

### ISSUE THREAD

**Issue by pathfinder-pf** [sgl-project/sglang-jax#338]
### Checklist

- [x] 1. I have searched related issues but cannot get the expected help.
- [x] 2. The bug has not been fixed in the latest version.
- [x] 3. Please note that if the bug-related issue you submitted lacks corresponding environment info and a minimal reproducible demo, it will be challenging for us to reproduce and resolve the issue, reducing the likelihood of receiving feedback.
- [x] 4. If the issue you raised is not a bug but a question, please raise a discussion at https://github.com/sgl-project/sgl-jax/discussions/new/choose Otherwise, it will be closed.
- [x] 5. Please use English, otherwise it will be closed.

### Describe the bug

when overlap-schedule by default, the performance decreases

### Reproduction

- launch server
```
jax==0.7.1 jaxlib==0.7.1 libtpu==0.0.17 flax==0.12.0
```
rm -rf /tmp/jit_cache
JAX_COMPILATION_CACHE_DIR=/tmp/jit_cache python3 -u -m sgl_jax.launch_server \
--model-path deepseek-ai/DeepSeek-R1-Distill-Qwen-1.5B \
--trust-remote-code  \
--tp-size=1 \
--device=tpu \
--mem-fraction-static=0.8 \
--chunked-prefill-size=2048 \
--download-dir=/tmp \
--dtype=bfloat16 \
--max-running-requests 64 \
--skip-server-warmup \
--page-size=64  \
--use-sort-for-toppk-minp \
--max-total-tokens=257536 \
--random-seed=27 \
--precompile-token-paddings=8192 \
--precompile-bs-paddings=64 

- eval
for ((i=0; i<64; i++));do
evalscope eval  --model deepseek-ai/DeepSeek-R1-Distill-Qwen-1.5B --api-url http://127.0.0.1:30000/v1/chat/completions --api-key EMPTY --eval-type service --datasets aime24  --eval-batch-size 64 --dataset-args '{"aime24":{"metric_list":["Pass@1"]}}' --generation-config '{"temperature": 0.6, "top_p": 0.95, "n":1, "max_tokens": 32768}' --timeout 120000
done
- result
avg 24.1
<img width="1125" height="454" alt="Image" src="https://github.com/user-attachments/assets/90e1e700-4844-4be3-9b0e-66a27b19bc8a" />

### Environment

v6e-1
